### PR TITLE
Fixed the bug in the metrics and switched to the UniformSample

### PR DESCRIPTION
**Pull request by lmineiro**
Big kudos to @vetinari for the `()` tip.
Also changed metrics reporting to match dropwizard's

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -69,20 +69,26 @@ func Init(o Options) {
 	reg = r
 }
 
+func createTimer() metrics.Timer {
+	return metrics.NewCustomTimer(metrics.NewHistogram(metrics.NewUniformSample(1024)), metrics.NewMeter())
```
Comment (aryszka) [inline metrics/metrics.go]: would be nice to have a name for this 1024

File: metrics/metrics.go [hunk 2]
```diff
@@ -69,20 +69,26 @@ func Init(o Options) {
...
+
 func getTimer(key string) metrics.Timer {
 	if reg == nil {
 		return nil
 	}
-	return reg.GetOrRegister(key, metrics.NewTimer()).(metrics.Timer)
+	return reg.GetOrRegister(key, createTimer).(metrics.Timer)
+}
+
+func updatetimer(key string, d time.Duration) {
```
Comment (mo-gr) [inline metrics/metrics.go]: please rename to `updateTimer`

**Comment by aryszka** [comment]
lgtm

**Comment by mo-gr** [comment]
lgtm

**Pull request merged**